lib/CfTemplateGenerators/CodeDeploy.js

```diff
@@ -38,6 +32,20 @@ function buildFnDeploymentGroup({ codeDeployAppName, deploymentSettings = {} })
       }
     }
   };
+
+  const lookupRole = {
+    'Fn::GetAtt': [
+      'CodeDeployServiceRole',
+      'Arn'
+    ]
+  }
+
+  if (typeof codeDeployRole != 'undefined') {
```

Comment:
This could be simplified with something like:

```javascript
const role =  codeDeployRole || lookupRole;
Object.assign(deploymentGroup.Properties, { SericeRoleArn: role });
```